### DISCUSSION TITLE
How to get a deep understanding on mlir infrastructure ?

### DISCUSSION THREAD

**Discussion by FlintWangacc** [iree-org/iree#18812] [General]
Recently, I have done some research on neural network compiler. 
I found the following software stack.
| Pytorch |
|:---------|
|TorchMlir|
|IREE        |
|MLIR       |
|LLVM      |
I tried to use pdb & gdb  to debug torch_mlir and iree-import-onnx to figure out the process that neural network transfer to mlir representation. But it seems some kind of cumbersome to me. I think the mlir infrastructure is very important. Is there some simple example that can help me to familiarize the mlir infrastructure?

**Comment by ScottTodd** [comment]
Some resources that may help:
* https://iree.dev/community/blog/2024-01-29-iree-mlir-linalg-tutorial/
* https://iree.dev/developers/general/developer-tips/
* https://mlir.llvm.org/docs/Tutorials/Toy/
* https://github.com/j2kun/mlir-tutorial
